src/ngraph/runtime/cpu/mkldnn_emitter.cpp

```diff
@@ -262,6 +265,188 @@ size_t MKLDNNEmitter::build_convolution_forward(const mkldnn::memory::desc& inpu
     return conv_index;
 }
 
+size_t MKLDNNEmitter::build_convolution(const ngraph::Node* node,
+                                        const std::vector<TensorViewWrapper>& args,
+                                        const std::vector<TensorViewWrapper>& out)
+{
+    auto convolution = static_cast<const ngraph::op::Convolution*>(node);
+    // For dilation, MKLDNN wants to know how many elements to insert between, not how far
+    // apart to space the elements like nGraph. So we have to subtract 1 from each pos.
+    ngraph::Strides window_dilation_strides_adjusted;
+    for (size_t s : convolution->get_window_dilation_strides())
+    {
+        window_dilation_strides_adjusted.push_back(s - 1);
+    }
+
+    auto input_format = mkldnn_utils::get_input_mkldnn_format(node, 0);
+    auto weights_format = mkldnn_utils::get_input_mkldnn_format(node, 1);
+    // HACK to help MKLDNN pick the right implementation
+    if (weights_format == mkldnn::memory::format::nchw)
+    {
+        weights_format = mkldnn::memory::format::oihw;
+    }
+    auto output_format = mkldnn_utils::get_output_mkldnn_format(node, 0);
+
+    auto input_data_desc = build_memory_descriptor(args[0], input_format);
+    auto weights_desc = build_memory_descriptor(args[1], weights_format);
+    auto result_desc = build_memory_descriptor(out[0], output_format);
+
+    return build_convolution_forward(input_data_desc,
+                                     weights_desc,
+                                     result_desc,
+                                     convolution->get_window_movement_strides(),
+                                     window_dilation_strides_adjusted,
+                                     convolution->get_padding_below(),
+                                     convolution->get_padding_above());
+}
+
+size_t MKLDNNEmitter::build_convolution_relu(const ngraph::Node* node,
+                                             const std::vector<TensorViewWrapper>& args,
+                                             const std::vector<TensorViewWrapper>& out)
+{
+    auto convolution = static_cast<const ngraph::op::ConvolutionRelu*>(node);
+
+    // For dilation, MKLDNN wants to know how many elements to insert between, not how far
+    // apart to space the elements like nGraph. So we have to subtract 1 from each pos.
+    Strides window_dilation_strides_adjusted;
```

Comment:
Can we have a function(build_conv) which does the initial stride check as well as getting the formats and building descriptors ? We can call that routine for every build_convolution_* rather than  duplicating that code in every build_convolution_*